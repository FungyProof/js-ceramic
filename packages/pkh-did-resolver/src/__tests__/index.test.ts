import index from '../index'
import { Resolver } from 'did-resolver'

const ethDid = 'did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb'
const btcDid = 'bdid:pkh:ip122:000000000019d6689c085ae165831e93:128Lkh3S7CkDTBZ8W7BbpsN3YYizJMp8p6'

describe('PKH DID Resolver', () => {
  it('successfully resolves DIDs', async () => {
    const resolverRegistry = index.getResolver()
    const resolver = new Resolver(resolverRegistry)


    expect(await resolver.resolve(ethDid)).toMatchSnapshot()
    expect(await resolver.resolve(ethDid)).toMatchSnapshot()

    expect(await resolver.resolve(ethDid, { accept: 'application/did+ld+json' })).toMatchSnapshot()
    expect(await resolver.resolve(ethDid, { accept: 'application/did+ld+json' })).toMatchSnapshot()
  })
})
