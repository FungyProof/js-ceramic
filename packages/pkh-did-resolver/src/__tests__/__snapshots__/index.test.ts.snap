// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PKH DID Resolver successfully resolves DIDs 1`] = `
Object {
  "didDocument": Object {
    "assertionMethod": Array [
      "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
    ],
    "authentication": Array [
      "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
    ],
    "id": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
    "verificationMethod": Array [
      Object {
        "blockchainAccountId": "eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
        "controller": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
        "id": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
        "type": "EcdsaSecp256k1RecoveryMethod2020",
      },
    ],
  },
  "didDocumentMetadata": Object {},
  "didResolutionMetadata": Object {
    "contentType": "application/did+json",
  },
}
`;

exports[`PKH DID Resolver successfully resolves DIDs 2`] = `
Object {
  "didDocument": Object {
    "assertionMethod": Array [
      "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
    ],
    "authentication": Array [
      "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
    ],
    "id": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
    "verificationMethod": Array [
      Object {
        "blockchainAccountId": "eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
        "controller": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
        "id": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
        "type": "EcdsaSecp256k1RecoveryMethod2020",
      },
    ],
  },
  "didDocumentMetadata": Object {},
  "didResolutionMetadata": Object {
    "contentType": "application/did+json",
  },
}
`;

exports[`PKH DID Resolver successfully resolves DIDs 3`] = `
Object {
  "didDocument": Object {
    "@context": Array [
      "https://www.w3.org/ns/did/v1",
      Object {
        "EcdsaSecp256k1RecoveryMethod2020": "https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020",
        "blockchainAccountId": "https://w3id.org/security#blockchainAccountId",
      },
    ],
    "assertionMethod": Array [
      "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
    ],
    "authentication": Array [
      "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
    ],
    "id": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
    "verificationMethod": Array [
      Object {
        "blockchainAccountId": "eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
        "controller": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
        "id": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
        "type": "EcdsaSecp256k1RecoveryMethod2020",
      },
    ],
  },
  "didDocumentMetadata": Object {},
  "didResolutionMetadata": Object {
    "contentType": "application/did+ld+json",
  },
}
`;

exports[`PKH DID Resolver successfully resolves DIDs 4`] = `
Object {
  "didDocument": Object {
    "@context": Array [
      "https://www.w3.org/ns/did/v1",
      Object {
        "EcdsaSecp256k1RecoveryMethod2020": "https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020",
        "blockchainAccountId": "https://w3id.org/security#blockchainAccountId",
      },
    ],
    "assertionMethod": Array [
      "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
    ],
    "authentication": Array [
      "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
    ],
    "id": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
    "verificationMethod": Array [
      Object {
        "blockchainAccountId": "eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
        "controller": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb",
        "id": "did:pkh:eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb#blockchainAccountId",
        "type": "EcdsaSecp256k1RecoveryMethod2020",
      },
    ],
  },
  "didDocumentMetadata": Object {},
  "didResolutionMetadata": Object {
    "contentType": "application/did+ld+json",
  },
}
`;
